docs/manuals/auth/authentication.md

```diff
@@ -0,0 +1,376 @@
+
+Beside their specific parameters, all authenticators have two parameters that you
+can provide in the [`Authenticator` constructor](Authenticator.__init__()^):
+
+- *secret_key*: a string that is used to encrypt the user credentials. Because
+  credentials are transmitted back and fourth when running Taipy on a server
+  (when a REST or a GUI application runs), this information is encrypted. You
+  can provide an encryption key, or let Taipy create one for you.
+- *auth_session_duration*: how long in seconds should the credentials created
+  by this authenticator be considered valid. 
+
+!!! Note "Global configuration"
+    If you are using Taipy Core (the `taipy.core` package), these parameters can also
+    be set in the [Taipy Global Configuration](../core/config/global-config.md).
+
+    The global configuration properties related to the default authentication
+    are used if the parameters are not loaded from a
+    *taipy&#x5F;auth&#x5F;&lt;protocol&gt;.json* file.
+
+    Here is how you cound use the _**Config.global_config.secret_key**_ and
+    _**Config.global_config.auth_session_duration**_ properties of the
+    global configuration:
+
+    === "Python configuration"
+
+        ```
+        Config.configure_global_app(secret_key = "my-ultra-secure-and-ultra-long-secret",
+                                    auth_session_duration = 600)  # 60 seconds = 10 minutes
+        ```
+
+    === "TOML configuration"
+
+        ```toml title="config.toml"
+
+        [TAIPY]
+        secret_key = "my-ultra-secure-and-ultra-long-secret"
+        auth_session_duration = 600   # 60 seconds = 10 minutes,
+        ```
+
+
+## 'None' Authenticator
+
+The *None* authenticator does not check for user declaration or password match.
+It is designed so that developers can start building secure applications before
+the actual authentication system and processes are defined or deployed.
+
+When the *None* authenticator `(Authenticator.)login()^` method is called, it always
+returns a valid `Credentials^` object, no matter what user name and password
+are provided.
+
+To create a *None* authenticator, you can instantiate an `Authenticator^` object
+setting the *protocol* argument of the constructor (or
+) to "none".
+
+## Taipy Authenticator
+
+The Taipy Authenticator is an internal authenticator originally designed for
+testing purposes, so an application can test authentication and authorization features
+without having to install and deploy a real authentication server.
+
+A Taipy Authenticator is created by the
+[`Authenticator` constructor](Authenticator.__init__()^) when invoked with the
+*protocol* argument set to "taipy".
+
+You can set the *roles* argument to a dictionary that associates a set of role
+names to every user name you want to grant login access to.<br/>
+Here is how you typically create a Taipy authenticator:
+```
+from taipy.auth import Authenticator
+roles={
+  "user1": "role1",
+  "user2": "role2",
+  "user3": ["role1", "role2"]
```

Comment:
I would also use some predefined roles to let the user know that he/she should populate the predefined roles as well.
Something like the following:

```suggestion
  "user1": ["role1", "TAIPY_READER"],
  "user2": ["role2", "TAIPY_ADMIN"],
  "user3": ["role1", "role2"m , "TAIPY_ADMIN"]
```

Reply:
Makes sense.